source/guide/3.2/reference-cli-tasks.md

```diff
@@ -0,0 +1,159 @@
+---
+layout: bt_wiki
+title: CLI Tasks Reference
+category: Reference
+publish: true
+abstract: "high level information on the Cloudify CLI tasks"
+pageord: 900
+---
+
+{%summary%}
+These tasks are used in [Manager blueprints](reference-terminology.html#manager-blueprints) to bootstrap a Cloudify Manager.
+This page will give a high level description of them.
+{%endsummary%}
+
+
+# bootstrap
+
+## Overview
+
+This bootstrap task uses the Fabric plugin to bootstrap a Cloudify Manager.
+
+It's currently a part of the CLI code, and can be found at / mapped to: `cloudify_cli.bootstrap.tasks.bootstrap`.
+
+The task does more or less the following:
+
+1) It connects to the Manager machine, downloads the various Cloudify packages and installs them.
+
+2) If supplied with one, it uploads an agent private key file to the Manager machine.
+
+3) It sets various runtime-properties on the *manager* node instance - this will later be used by the CLI to extract data in order to set the configuration for the local environment.
+
+4) It uploads the [*provider context*](reference-terminology.html#provider-context) object to the Manager via the Manager's REST service.
+
+
+## API
+
+The bootstrap task takes several parameters:
+
+  * ***cloudify_packages*** - This is the only required parameter. It points at links of Cloudify packages to be installed on the Manager, and it should be in the same schema as the *manager*'s node `cloudify_packages` property. See the reference for the [CloudifyManager type](reference-types.html#cloudifymanager-type) for more information.
+
+  * ***agent_local_key_path*** - Optional parameter. Sets the agent private key local file path. This file will be uploaded to the Manager machine and be used by default for setting up and connecting to Cloudify agent hosts. If omitted, no agent key file will be uploaded to the Manager machine.
+
+  * ***agent_remote_key_path*** - Optional parameter. Sets the agent private key remote file path. This is the path on the manager to which the agent private key file will be uploaded. If this parameter is omitted, the file will sit at `~/.ssh/agent_key.pem`.
+
+  * ***manager_private_ip*** - Optional parameter. This is the IP that will be used by agent hosts to communicate with the Manager machine (it should be the private IP since they communicate via an internal network). If this parameter is omitted, the *manager*'s node instance's `host_ip` will be used instead.
+
```

Comment:
well this issue isn't really covered. it depends on the plugin. host_ip extracts the private ip from the node runtime properties. so its the plugin's responsibility to set this ip with the correct one (the one from the management subnet in case of openstack). this is explained in the openstack plugin